compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -417,6 +472,50 @@ impl<'a> Parser<'a> {
                 return Ok(recovered);
             }
         }
+        if suggest_removing {
+            err.span_suggestion(
+                self.token.span,
+                format!(
+                    "Consider removing {}",
+                    match self.token.kind {
+                        TokenKind::Lt
```

Comment:
Yea, there already is one, but it is in rustc_ast_pretty/src/pprust/state.rs as a private method and it also takes an additional `convert_dollar_crate: Option<Span>` should i copy that code into a method for tokenkind and reuse it there and for this diagnostic's purpose? (ofc i'd have to remove this convert_dollar_crate parameter somehow but it is only used once in this fn so that shouldn't be a problem)

Reply:
I think this is fine.